### DISCUSSION TITLE
How to build libVTMDecoder?

### DISCUSSION THREAD

**Discussion by frankplow** [IENT/YUView#527] [Q&A] [answered]
I'm trying to build libVTMDecoder, but can't seem to find any shim source file to adapt from the libDecoderLib library produced when compiling VTM to the libVTMDecoder expected by this project. I've tried compiling VTM from both the [official repo](https://vcgit.hhi.fraunhofer.de/jvet/VVCSoftware_VTM) and the [GitHub-hosted mirror](https://github.com/ChristianFeldmann/VTM) provided by @ChristianFeldmann. The only other search result on GitHub for VTMDecoder is [VTMDecoder_VLCPlugin](https://github.com/InterDigitalInc/VTMDecoder_VLCPlugin), but this seems to build a library called vvcdecoder_plugin. There's also the libvtmdec provided by old versions of VTM, but again this doesn't appear to be the right library.

What am I missing here? Thanks for any help.

Think this may be related to #227 and #302.

**Comment by ChristianFeldmann** [comment, marked as answer]
Hi! By default, the reference decoder VTM does not come with a library interface but as a standalone application. So I added one. It was never merged into the main branch. You can find the code to compile the library in the `decoderLib` branch of may VTM fork: https://github.com/ChristianFeldmann/VTM/tree/feature/decoderLib
But be aware that this code is very outdated and could use some update from remote.

**Reply by frankplow** [reply]
I hadn't spotted that branch - that was the missing piece of the puzzle. Thanks a lot.
